### PR TITLE
 Handle response from HMI during resumption data

### PR DESCRIPTION
**Pull request by LuxoftAKutsan**
The aim of this feature is to handle HMI error responses while restoring the data during the resumption process.

### CODE REVIEW

**Comment by LuxoftAKutsan** [comment]
@theresalech  this PR is ready for review

File: proposals/nnn-resumption-data-error-handling.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+
+#### Do not check response from HMI
+If application was able to send any persistent data during previous connection, there should be no reason not to restore this data.  
+So this proposal is more about unexpected or incorrect HMI behavior. 
+
+
+#### Create special result code "partial resumption"
+
+Prevent failing resumption in case when part of the data was rejected by HMI, and send in RegisterAppInterfaceResponse information what data were not restored.
```
Comment (theresalech) [inline proposals/nnn-resumption-data-error-handling.md]: Can you please reword?  I'm not entirely sure what's intended here, but perhaps "Prevent resumption failing in the case when only part of the data is rejected by the HMI, then send in the RAIR info field what data was not able to be restored" ?

**Comment by LuxoftAKutsan** [comment]
@theresalech all review comments fixed.

File: proposals/nnn-resumption-data-error-handling.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+ - DeleteInteractionChoiceSet
+ - ResetGlobalProperties
+ - UnsubscribeButton
+ - UnsubscribeVehicleData
+ - UnsubscribeWayPoints
+
+In case if some data like a subscription already used by other applications, it means that subscription is actual and SDL should not send unsubscribe requests to HMI. 
+
+
+After reverting persistent data SDL should response `RegisterAppInterfaceResponse(success=true,result_code=RESUME_FAILED)` to mobile application.
```
[the remark this answers is not in the document]
Comment (theresalech) [reply proposals/nnn-resumption-data-error-handling.md]: This still needs to be updated.

File: proposals/nnn-resumption-data-error-handling.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+If multiple applications trying to restore the same subscription SDL should send the only first subscription to HMI.
```
[the remark this answers is not in the document]
Comment (theresalech) [reply proposals/nnn-resumption-data-error-handling.md]: This still needs to be updated.

File: proposals/nnnn-resumption-data-error-handling.md [hunk 4]
```diff
@@ -66,13 +66,13 @@ Impacts resumption component of SDL and application registration process.
...
 #### Do not check response from HMI
-If application was able to send any persistent data during previous connection, there should be no reason not to restore this data.  
+If application was able to send any persistent data during previous connection, there should be no reason to not restore this data.  
 So this proposal is more about unexpected or incorrect HMI behavior. 
 
 
 #### Create special result code "partial resumption"
 
-Prevent failing resumption in case when part of the data was rejected by HMI, and send in RegisterAppInterfaceResponse information what data were not restored. 
+Prevent resumption failing in the case when part of the data is rejected by the HMI, then send in the RegisterAppInterfaceResponse information what data were not restored.
```
Comment (theresalech) [inline proposals/nnnn-resumption-data-error-handling.md]: Please update to "...information field what data was not able to be restored."

**Comment by LuxoftAKutsan** [comment]
@theresalech done

File: proposals/nnnn-resumption-data-error-handling.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+ - DeleteInteractionChoiceSet
+ - ResetGlobalProperties
+ - UnsubscribeButton
+ - UnsubscribeVehicleData
+ - UnsubscribeWayPoints
+
+In the case some data, like a subscription, is already used by other applications, this means that the subscription is actual and SDL should not send unsubscribe requests to HMI. 
+
+
+After reverting persistent data SDL should response `RegisterAppInterfaceResponse(success=true,result_code=RESUME_FAILED)` to mobile application.
```
Comment (theresalech) [inline proposals/nnnn-resumption-data-error-handling.md]: This seems to keep getting missed.  Please update "response" to "respond"

File: proposals/nnnn-resumption-data-error-handling.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+If multiple applications trying to restore the same subscription SDL should send the only first subscription to HMI.
```
Comment (theresalech) [inline proposals/nnnn-resumption-data-error-handling.md]: This also seems to keep getting missed.  Please update to "If multiple applications are trying to restore the same subscription, SDL should..."

**Comment by LuxoftAKutsan** [comment]
@theresalech proposal updated

**Pull request merged**